backend/drm/drm.c

```diff
@@ -598,6 +598,8 @@ static bool wlr_drm_output_set_cursor(struct wlr_output *_output,
 		wlr_matrix_texture(plane->matrix, plane->width, plane->height,
 			output->output.transform ^ WL_OUTPUT_TRANSFORM_FLIPPED_180);
 
+		// TODO the image needs to be rotated depending on the output rotation
```

Comment:
I've been working on this for awhile and I don't yet have a good explanation for this bug. Rotating the image will not solve the problem like I thought it would.

For a rotated monitor, the cursor image must be rotated in some 90 degree angle increment. But whenever the output transform is given to rotate 90 or 270 degrees, it puts the cursor in the wrong position on the screen translated about 100 pixels to the left or right. If I try to translate it to the right position, it seems to go out of bounds of some clipping box.

Reply:
The buffer that the cursor sits on (the DRM plane) may be much larger than the cursor actually is. The cursor is sitting in the top-left corner.
On some intel hardware I own, the cursor plane is 256x256 (although 64x64 is more typical for other hardware), so just naively rotating the cursor is going to put it in the wrong place by potentially quite a long way.

You're going to need to translate by the plane dimensions. The cursor plane dimensions are found in output->crtc->cursor->{width,height}.